apps/wallet-dashboard/app/(protected)/home/page.tsx

```diff
@@ -2,19 +2,53 @@
 // SPDX-License-Identifier: Apache-2.0
 'use client';
 
-import { AccountBalance, MyCoins, TransactionsOverview, StakingOverview } from '@/components';
+import {
+    AccountBalance,
+    MyCoins,
+    TransactionsOverview,
+    StakingOverview,
+    MigrationOverview,
+} from '@/components';
+import { useGetCurrentEpochStartTimestamp } from '@/hooks';
+import { groupStardustObjectsByMigrationStatus } from '@/lib/utils';
 import { useFeature } from '@growthbook/growthbook-react';
-import { Feature } from '@iota/core';
+import {
+    Feature,
+    STARDUST_BASIC_OUTPUT_TYPE,
+    STARDUST_NFT_OUTPUT_TYPE,
+    useGetAllOwnedObjects,
+} from '@iota/core';
 import { useCurrentAccount, useCurrentWallet } from '@iota/dapp-kit';
 import clsx from 'clsx';
 
 function HomeDashboardPage(): JSX.Element {
     const { connectionStatus } = useCurrentWallet();
     const account = useCurrentAccount();
 
+    const address = account?.address || '';
+    const { data: currentEpochMs } = useGetCurrentEpochStartTimestamp();
+    const { data: basicOutputObjects } = useGetAllOwnedObjects(address, {
+        StructType: STARDUST_BASIC_OUTPUT_TYPE,
+    });
+    const { data: nftOutputObjects } = useGetAllOwnedObjects(address, {
+        StructType: STARDUST_NFT_OUTPUT_TYPE,
+    });
+    const { migratable: migratableBasicOutputs } = groupStardustObjectsByMigrationStatus(
+        basicOutputObjects ?? [],
+        Number(currentEpochMs),
+        address,
+    );
+
+    const { migratable: migratableNftOutputs } = groupStardustObjectsByMigrationStatus(
+        nftOutputObjects ?? [],
+        Number(currentEpochMs),
+        address,
+    );
+
     const stardustMigrationEnabled = useFeature<boolean>(Feature.StardustMigration).value;
-    // Add the logic here to check if the user has migration objects.
-    const needsMigration = false && stardustMigrationEnabled;
+    const needsMigration =
+        (migratableBasicOutputs.length > 0 || migratableNftOutputs.length > 0) &&
+        stardustMigrationEnabled;
```

Comment:
we could first check if the `stardustMigration` is enabled, and if so, check if we have outputs to migrate. This way we avoid making unnecessary queries